kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java

```diff
@@ -140,6 +159,10 @@ public void destroy(KieServerImpl kieServer, KieServerRegistry registry) {
         if (deploymentService != null) {
             deploymentService.removeListener(this);
         }
+        if (producerReady.compareAndSet(true, false)) {
+            producer.close();
```

Comment:
Shouldn't we try to `flush` first? or right after sending a record? wdyt?
```suggestion
            producer.flush();
            producer.close();
```

Reply:
as far as I know, close internally invoke flush (in reality it waits for the acks till a timeout is reach) so it will be redudant to call flush here

Reply:
I see, thanks for that